theano/compile/builders.py

```diff
@@ -133,6 +135,59 @@ def perform(self, node, inputs, outputs):
             # we wont need this copy anymore
             output[0] = variable.copy()
 
+    def connection_pattern(self, node):
```

Comment:
What I implemented is indeed the ```thunks``` that using dimensions of variables as inputs and outputs to calculate to shape. This allow to reuse the optimized fgraph and bypass the optimization. So it is similar to every ```make_thunk``` method in every ```Linker``` and ```ScanOp```, if you are talking about this. Maybe I can abstract the code that go over the fgraph and do the calculation, as I have similar operation to each node.

I cannot find any ``` connection_pattern``` in ```ScanOp```.